target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -236,6 +237,11 @@ ipq40xx_setup_macs()
 	pakedge,wr-1)
 		wan_mac=$(macaddr_add $(get_mac_label) 1)
 		;;
+	tel,x1pro)
+		label_mac=$(mtd_get_mac_binary "0:ART" 0x1006)
```

Comment:
This still doesn't make sense as if its just in binary form then NVMEM must work for reading the MAC

Reply:
I still dont like this, NVMEM just must work if its just stored in binary

Reply:
@neheb Do you have an idea why this fails from NVMEM, I cant see why it would fail and really, really dont want more userspace MAC fetching.

Reply:
Is the mac address failing for all ethernet ports or just the WAN?

Reply:
Just ran kernel_oldconfig. I saw

```
+CONFIG_NVMEM_LAYOUTS=y
+CONFIG_NVMEM_LAYOUT_U_BOOT_ENV=y
```

Not sure if related. I also saw a bunch of other changes that I don't really want to touch.

Reply:
Maybe it's caused by syntax error?
`nvmem-cell-names = "pre-calibration, mac-address";` => `nvmem-cell-names = "pre-calibration", "mac-address";`

@mips171 Please patch NVMEM mac address changes to a new commit and push it so that we can better check it.

Reply:
@mips171 any update on this?